.github/workflows/core-network-services-deployment.yml

```diff
@@ -0,0 +1,55 @@
+---
+name: core-network-services-deployment
+
+on: 
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+    paths:
+      - 'terraform/environments/core-network-services/**'
+  workflow_dispatch:
+
+env:
+  AWS_ACCESS_KEY_ID:  ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY:  ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  TF_IN_AUTOMATION: true
+
+jobs:
+  core-network-services-deployment-plan:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'pull_request'
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Terraform plan
```

Comment:
You'll need to add a `terraform init` here otherwise it'll fail because of uninstalled providers 👍🏼

Reply:
ok - will add it to the yaml steps